Feature/auth logic

AnastasiaLes opened the pull request:
1. User can signup and login (connected to the backend)
![Screenshot 2022-11-11 180611](https://user-images.githubusercontent.com/87652792/201417870-b0b8d7a4-6c35-459b-9150-795b2647d46e.jpg)
![Screenshot 2022-11-11 180510](https://user-images.githubusercontent.com/87652792/201417886-ee3b3498-0ee4-46af-8969-07b539a38e72.jpg)


Fixed:
1. linter checked

AnastasiaLes committed:
Login form added

AnastasiaLes committed:
login form fixed

AnastasiaLes committed:
Sign Up Form Added

AnastasiaLes committed:
fix

AnastasiaLes committed:
auth-slice fix

AnastasiaLes committed:
login setup

AnastasiaLes committed:
Squashed commit of the following:

commit 850db097e7dc6702edb550226286b37459a64629
Author: Pavlo Chernykh <95277626+pavlo-chernykh@users.noreply.github.com>
Date:   Fri Nov 11 01:19:35 2022 +0200

    Update app.tsx

commit 4ba9e571db902484033f704e5a71ef0e12079301
Merge: 53fe042 b7b70f3
Author: Pavlo Chernykh <95277626+pavlo-chernykh@users.noreply.github.com>
Date:   Fri Nov 11 01:12:05 2022 +0200

    Merge pull request #5 from ZenBit-Tech/feature/profile-questions-1

    feature/profile questions 1

commit b7b70f3507375b5b6fe84ca1542debafb4d5d72f
Merge: f5b6420 53fe042
Author: Pavlo Chernykh <95277626+pavlo-chernykh@users.noreply.github.com>
Date:   Fri Nov 11 01:11:21 2022 +0200

    Merge branch 'dev' into feature/profile-questions-1

commit 53fe042ac549f2e46b344dae7aff93070479bb16
Merge: 252047a 592f298
Author: Mykhailo Kutsyi <91030243+MykhailoKutsyi@users.noreply.github.com>
Date:   Fri Nov 11 00:28:32 2022 +0200

    Merge pull request #14 from ZenBit-Tech/feature/globalstyle

    Feature/globalstyle

commit 252047af486c7e444d725d87ae30f31fc083d04b
Merge: b39925a 295a0c8
Author: Mykhailo Kutsyi <91030243+MykhailoKutsyi@users.noreply.github.com>
Date:   Fri Nov 11 00:28:04 2022 +0200

    Merge pull request #11 from ZenBit-Tech/check/lint

    Check/lint

commit b39925a9cbc56b6f752146d0fa3a686a0a78fa75
Merge: 822d538 c29e9c1
Author: Mykhailo Kutsyi <91030243+MykhailoKutsyi@users.noreply.github.com>
Date:   Fri Nov 11 00:27:50 2022 +0200

    Merge pull request #8 from ZenBit-Tech/fix/lint

    Fix/lint

commit 592f2986e1abbcc8c613e0f30a429ba5126544c3
Author: MykhailoKutsyi <91030243+MykhailoKutsyi@users.noreply.github.com>
Date:   Fri Nov 11 00:03:53 2022 +0200

    add GlobalStyle

commit 295a0c8747e69e43fcd7fc671ad439810ea161a2
Author: MykhailoKutsyi <91030243+MykhailoKutsyi@users.noreply.github.com>
Date:   Thu Nov 10 01:26:55 2022 +0200

    update eslint script again

commit cb952fa4a5124a7b3c52ead76e8cacba568878c0
Author: MykhailoKutsyi <91030243+MykhailoKutsyi@users.noreply.github.com>
Date:   Thu Nov 10 01:19:13 2022 +0200

    update eslint script

commit f9d32e75d78bce26e7a3d90b69fdec8e6c99c733
Author: MykhailoKutsyi <91030243+MykhailoKutsyi@users.noreply.github.com>
Date:   Thu Nov 10 01:17:29 2022 +0200

    check it it it it

commit a286f0ed196c294dba879297c42d5ce111c5c02e
Author: MykhailoKutsyi <91030243+MykhailoKutsyi@users.noreply.github.com>
Date:   Thu Nov 10 01:15:08 2022 +0200

    check it it it

commit 25335573a10e4c19e69a723cfab46546c3b29e67
Author: MykhailoKutsyi <91030243+MykhailoKutsyi@users.noreply.github.com>
Date:   Thu Nov 10 01:08:53 2022 +0200

    check it it

commit f5b642036633876a0cbe66f2632f38d356fb8249
Author: Pavlo Chernykh <pcherny10@gmail.com>
Date:   Wed Nov 9 23:01:31 2022 +0200

    fixed sufix

commit d77c2ae98f2b3c98cab487b6f0792a5dfb857d83
Author: Pavlo Chernykh <pcherny10@gmail.com>
Date:   Wed Nov 9 22:52:42 2022 +0200

    fixes2

commit b3a377e758a33ba3f5b0bd572059aa85843295f7
Author: Pavlo Chernykh <pcherny10@gmail.com>
Date:   Wed Nov 9 15:58:06 2022 +0200

    test123

commit 14c6a77061f14ed0004527edec1e2077957149ae
Author: Pavlo Chernykh <pcherny10@gmail.com>
Date:   Wed Nov 9 15:42:33 2022 +0200

    test lint2

commit 95f49d1e45ea9499cfc3499d42141ca01f6b1729
Author: Pavlo Chernykh <pcherny10@gmail.com>
Date:   Wed Nov 9 14:49:59 2022 +0200

    test312

commit c67651372494a1f41ffbec99f5bb4986f21c557c
Author: Pavlo Chernykh <pcherny10@gmail.com>
Date:   Wed Nov 9 14:46:08 2022 +0200

    test2

commit 653cc8f1663d8c88a3fe166cb4b1d401a950a61f
Author: Pavlo Chernykh <pcherny10@gmail.com>
Date:   Wed Nov 9 14:39:41 2022 +0200

    test

commit 7468098b7ea74d3abd723f764fc8e3782d0502d6
Author: Pavlo Chernykh <pcherny10@gmail.com>
Date:   Wed Nov 9 14:37:57 2022 +0200

    test

commit 38403210158edfa64658454c61df9a88677bd78f
Merge: c29e9c1 957f50a
Author: Pavlo Chernykh <pcherny10@gmail.com>
Date:   Wed Nov 9 14:28:00 2022 +0200

    changes

commit 957f50ad7bb05f4b45feb4146108f639d7b875ba
Author: Pavlo Chernykh <pcherny10@gmail.com>
Date:   Wed Nov 9 11:24:56 2022 +0200

    fixed imports according to rules

commit 2ddb30e0bd91d5cf3ffd2b172b856f748bc83669
Author: Pavlo Chernykh <pcherny10@gmail.com>
Date:   Wed Nov 9 11:19:28 2022 +0200

    added constants folder and added fixes

commit c29e9c1977447d793dcf8efc63211790bc0166f3
Author: MykhailoKutsyi <91030243+MykhailoKutsyi@users.noreply.github.com>
Date:   Wed Nov 9 05:27:03 2022 +0200

    check how its works

commit d239a4301a0cef5522a635db44aa7703b39b3c5d
Author: MykhailoKutsyi <91030243+MykhailoKutsyi@users.noreply.github.com>
Date:   Wed Nov 9 04:10:24 2022 +0200

    format prettier all files

commit 82e6b1527d8e034f669388c51dd53de92b671a45
Author: MykhailoKutsyi <91030243+MykhailoKutsyi@users.noreply.github.com>
Date:   Wed Nov 9 04:07:34 2022 +0200

    upd

commit 7541fb0cf7ba6a8b6e3d0f6046576269ddada6b5
Author: MykhailoKutsyi <91030243+MykhailoKutsyi@users.noreply.github.com>
Date:   Wed Nov 9 04:05:50 2022 +0200

    update prettier

commit 7b8644916c051cb212010275ade8e7ca50f0b86f
Author: MykhailoKutsyi <91030243+MykhailoKutsyi@users.noreply.github.com>
Date:   Wed Nov 9 03:52:31 2022 +0200

    update absolute path

commit d87eba4776216b2ac72830bd348be1b1018e2bf9
Author: MykhailoKutsyi <91030243+MykhailoKutsyi@users.noreply.github.com>
Date:   Wed Nov 9 01:56:48 2022 +0200

    commitos

commit e2738c9c842b21394ddcf12d96ae1ebbc93b5a7b
Author: MykhailoKutsyi <91030243+MykhailoKutsyi@users.noreply.github.com>
Date:   Wed Nov 9 01:55:37 2022 +0200

    commita

commit 313f8dc330c83b5263d9abee415f4fd4b128ccbf
Author: MykhailoKutsyi <91030243+MykhailoKutsyi@users.noreply.github.com>
Date:   Wed Nov 9 01:00:18 2022 +0200

    check formatr

commit 5d35622c0933434b639d22b4da2e680a1a7d5d53
Author: MykhailoKutsyi <91030243+MykhailoKutsyi@users.noreply.github.com>
Date:   Wed Nov 9 00:49:22 2022 +0200

    check formating

commit 8ccf35673a8fd409feef7e0e536cb93cfa006e2b
Author: MykhailoKutsyi <91030243+MykhailoKutsyi@users.noreply.github.com>
Date:   Wed Nov 9 00:42:23 2022 +0200

    check format

commit f8fd61c62793ffc8e1f2412a4a74969438cd96b5
Author: MykhailoKutsyi <91030243+MykhailoKutsyi@users.noreply.github.com>
Date:   Wed Nov 9 00:20:06 2022 +0200

    test

commit 3901c46b88ff7c9342f06795c2f4d7b653508797
Author: MykhailoKutsyi <91030243+MykhailoKutsyi@users.noreply.github.com>
Date:   Wed Nov 9 00:14:24 2022 +0200

    Keep calm and commit

commit 9a107c105050cb3a0cebcbfa9e10f28ca36bdc72
Author: MykhailoKutsyi <91030243+MykhailoKutsyi@users.noreply.github.com>
Date:   Tue Nov 8 23:52:56 2022 +0200

    configure husky

commit effd01f076f5fe4474070f503c76ebdb60508075
Author: MykhailoKutsyi <91030243+MykhailoKutsyi@users.noreply.github.com>
Date:   Tue Nov 8 23:52:09 2022 +0200

    configure prettier, husky and lint-staged

commit 8fc868abb8ba52526a7441ac40e221518c558052
Author: MykhailoKutsyi <91030243+MykhailoKutsyi@users.noreply.github.com>
Date:   Tue Nov 8 23:45:33 2022 +0200

    conf husky

commit 3fe80c90c01de270cbe55850c0c85af42a9a1ca2
Author: Pavlo Chernykh <pcherny10@gmail.com>
Date:   Tue Nov 8 20:13:03 2022 +0200

    added input component, route alias, form + some form items

commit 739fc895093812edd900f747997eb415f79fed20
Author: Pavlo Chernykh <pcherny10@gmail.com>
Date:   Tue Nov 8 17:17:21 2022 +0200

    added alias and input

commit 9a1487749a14524bcad58af19a827adc5f6d0dc4
Author: Pavlo Chernykh <pcherny10@gmail.com>
Date:   Tue Nov 8 15:06:15 2022 +0200

    added path alias

commit 6418d22c5c7c0ff384323c5da4c71e58b7974360
Author: Pavlo Chernykh <pcherny10@gmail.com>
Date:   Tue Nov 8 14:38:17 2022 +0200

    added

commit c2cde90f068964980fcd96aa44c9d378eb1eb004
Merge: a25889c 822d538
Author: Pavlo Chernykh <pcherny10@gmail.com>
Date:   Tue Nov 8 14:20:13 2022 +0200

    Merge https://github.com/ZenBit-Tech/coffee_lovers_frontend into feature/profile-questions-1

commit a25889c7f0e7343f4adce0578128ff42af6bfdd1
Merge: c665c12 7e89f24
Author: Pavlo Chernykh <pcherny10@gmail.com>
Date:   Mon Nov 7 22:46:50 2022 +0200

    fixed conflict

commit 822d53882273b239abc235619f2ae493f665e5bd
Merge: 5811293 7e89f24
Author: Liudmyla <lyuda.dzyubinska@gmail.com>
Date:   Tue Nov 8 02:43:20 2022 +0600

    Merge pull request #1 from ZenBit-Tech/feature/i18n

    Feature/i18n

commit c665c126db7687257c182064c3c93f562e3bc109
Author: Pavlo Chernykh <pcherny10@gmail.com>
Date:   Mon Nov 7 22:17:49 2022 +0200

    added progress bar and routing

commit 2d2b02fba9104d513ee6cd1a7a0fa2f919b48b2c
Author: Pavlo Chernykh <pcherny10@gmail.com>
Date:   Mon Nov 7 19:02:29 2022 +0200

    test

AnastasiaLes committed:
auth logic

AnastasiaLes committed:
auth logic setup

AnastasiaLes committed:
conflicts fix

lyudad commented on apps/freelance/src/redux/auth/auth-api.ts:
```diff
@@ -0,0 +1,33 @@
+
+export const authApi = createApi({
+  reducerPath: 'authApi',
+  baseQuery: fetchBaseQuery({ baseUrl: 'http://localhost:5000/auth/' }),
```
today you have localhos, tomorrow https://mysuperwebsite/ you will all the time change URL?
Please read what it env variables are, why we need to use it, and when. Create the list for me: in which case you will use env, in which constants, and when you will write something directly in the code, and send it to me in private message, without this work I will not accept your PR)

AnastasiaLes replied:
done

AnastasiaLes replied:
![image](https://user-images.githubusercontent.com/87652792/201671121-90eecd38-4153-4ec8-8651-a661389de3a8.png)

lyudad commented on apps/freelance/src/redux/auth/auth-slice.ts:
```diff
@@ -0,0 +1,41 @@
+  initialState,
+  reducers: {
+    setUser(state, action: PayloadAction<{ name: string; token: string }>) {
+      localStorage.setItem(
```
no, there is redux persist library which will handle it instead of you

AnastasiaLes replied:
persist library added

AnastasiaLes committed:
fix auth logic

lyudad commented on apps/freelance/src/redux/auth/auth-api.ts:
```diff
@@ -0,0 +1,59 @@
+
+import { RootState } from '../store';
+// import { persistReducer } from 'redux-persist';
+// import storage from 'redux-persist/lib/storage';
```
you have not checked git checklist. it should be checked before write me a message that you are ready
<img width="819" alt="Снимок экрана 2022-11-14 в 21 49 33" src="https://user-images.githubusercontent.com/25103090/201762847-2435c23d-91be-4b41-8e35-6caba17707ac.png">

AnastasiaLes replied:
read the guide a couple of times and cleared the code

lyudad commented:
I will check the rest only after you will confirm that you went through your PR and checklist and no issue have been found

AnastasiaLes committed:
issues fix

AnastasiaLes committed:
persist fix

AnastasiaLes committed:
redux-persisr fix

AnastasiaLes committed:
conflicts fix

AnastasiaLes committed:
fixes

AnastasiaLes committed:
fix

AnastasiaLes committed:
Checkbox and errors fix

AnastasiaLes committed:
errors fix

AnastasiaLes committed:
fix

AnastasiaLes committed:
Delete .env.local

AnastasiaLes commented:
I added all up-to-date fixes in this PR:
1. checkbox added to signup form
2. all errors were added and styled
3. fixed signup logic
![image](https://user-images.githubusercontent.com/87652792/202292224-bcfb29c4-3351-42e1-a5eb-804ee744009f.png)

lyudad commented on apps/freelance/src/pages/LoginPage/styles.ts:
```diff
@@ -0,0 +1,8 @@
+  margin-top: 50px;
+  margin-left: 100px;
+`;
+export const Title = styled.h1``;
```
<img width="744" alt="Снимок экрана 2022-11-15 в 20 29 24" src="https://user-images.githubusercontent.com/25103090/202296831-c1812155-06c1-4d7d-b19a-5298c63f69d0.png">

AnastasiaLes replied:
fixed

lyudad commented on libs/components/src/lib/login-form/login-form.tsx:
```diff
@@ -0,0 +1,115 @@
+
+import { StyledInput, StylesButton } from './styles';
+
+import 'antd/dist/antd.css';
```
it should be only once in index file of root

AnastasiaLes replied:
done

lyudad replied:
I still see it here

lyudad commented on libs/components/src/lib/login-form/login-form.tsx:
```diff
@@ -0,0 +1,115 @@
+
+  const goToReset = () => {
+    navigate('/passwordreset');
+  };
```
too many logic, create custom hook

AnastasiaLes replied:
fixed

lyudad commented on libs/components/src/lib/signup-form/signup-form.tsx:
```diff
@@ -0,0 +1,166 @@
+
+import { StyledInput, StylesButton } from '../login-form/styles';
+
+import 'antd/dist/antd.css';
```
it should be one import for full project

AnastasiaLes replied:
done

lyudad replied:
what done, why I still see it here

AnastasiaLes replied:
@lyudad I removed StyledInput and StyledButton from both files, didn't understand that you meant antd.css

lyudad commented on libs/components/src/lib/signup-form/signup-form.tsx:
```diff
@@ -0,0 +1,166 @@
+    .required()
+    .oneOf([yup.ref('password')]),
+  agreement: yup.bool().oneOf([true]),
+});
```
it should be in separate file

AnastasiaLes replied:
fixed

AnastasiaLes committed:
fixes

AnastasiaLes committed:
fix

AnastasiaLes committed:
Merge branch 'feature/authLogic' of https://github.com/ZenBit-Tech/coffee_lovers_frontend into feature/authLogic

AnastasiaLes committed:
fix

AnastasiaLes committed:
gitignore fix

AnastasiaLes committed:
Merge branch 'dev' of https://github.com/ZenBit-Tech/coffee_lovers_frontend into dev

AnastasiaLes committed:
store fix

AnastasiaLes committed:
error fix

AnastasiaLes committed:
fix

AnastasiaLes committed:
Merge branch 'dev' of https://github.com/ZenBit-Tech/coffee_lovers_frontend into dev

AnastasiaLes committed:
fix
